lib/git-notes.ts

```diff
@@ -76,12 +73,8 @@ export class GitNotes {
     }
 
     public async update(url: string): Promise<void> {
-        if (this.notesRef === "refs/notes/gitgitgadget" ||
-            this.notesRef === "refs/notes/commit-to-mail" ||
-            this.notesRef === "refs/notes/mail-to-commit") {
-            await git(["fetch", url,
-                       `+${this.notesRef}:${this.notesRef}`],
-                      { workDir: this.workDir });
+        if (this.notesRef.match(/^refs\/notes\/(gitgitgadget|commit-to-mail|mail-to-commit)$/)) {
```

Comment:
An alternative would be `["refs/notes/gitgitgadget", "refs/notes/commit-to-mail", "refs/notes/mail-to-commit"].includes(this.notesRef)`. But the regex is fine, too (despite [some concerns](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/), this seems outside [dangerous territory](https://stackstatus.tumblr.com/post/147710624694/outage-postmortem-july-20-2016)).

Reply:
I am adding [eslint-plugin-security](https://github.com/eslint-community/eslint-plugin-security#readme) to replace the trojan attack detection.  It also includes unsafe regex checking.